### PR TITLE
add PUT /bundles api.

### PR DESCRIPTION
**Pull request by ttung**
Running the unit test writes the following bundle to the store:

```
{"format": "0.0.1", "version": "2017-06-20T21:41:25.793628Z", "files": [{"name": "LICENSE", "uuid": "f3926590-5f38-429c-a024-63bab88e97fe", "version": "2017-06-20T21:41:23.811824Z", "content-type": "text/plain", "indexed": false, "crc32c": "e16e07b9", "s3-etag": "3b83ef96387f14655fc854ddc3c6bd57", "sha1": "2b8b815229aa8a61e483fb4ba0588b8b6c491890", "sha256": "cfc7749b96f63bd31c3c42b5c471bf756814053e847c10f3eb003417bc523d30"}], "creator_uid": 12345}
```

### CODE REVIEW

File: dss-api.yml [hunk 1]
```diff
@@ -258,6 +258,79 @@ paths:
...
           schema:
             $ref: '#/definitions/Error'
+    put:
+      summary: Create a bundle
+      description: |
+        Create a new bundle with a given UUID.  The list of files UUID+versions to be included must be provided.
+      parameters:
+        - name: uuid
+          in: path
+          description: A RFC4122-compliant ID for the file.
```
Comment (kislyuk) [inline dss-api.yml]: s/file/bundle/?

File: dss-api.yml [hunk 2]
```diff
@@ -258,6 +258,79 @@ paths:
...
+          description: Timestamp of bundle creation in RFC3339.
+          required: false
+          type: string
+          format: date-time
+        - name: replica
+          in: query
+          description: Replica to write to.
+          required: true
+          type: string
+          pattern: "^(AWS|GC|Azure)$"
```
Comment (kislyuk) [inline dss-api.yml]: Does the spec say how this should be formatted? I was using `(aws|gcs|azure)` elsewhere, we should make sure they are consistent.

Also, I believe swagger has enums, up to you if you want to use it.

File: dss-api.yml [hunk 3]
```diff
@@ -258,6 +258,79 @@ paths:
...
+                      description: Timestamp of file creation in RFC3339.
+                      type: string
+                      format: date-time
+                    name:
+                      description: Name of the file.
+                      type: string
+                      pattern: "^[^/\\\\]+$"
+                    indexed:
+                      description: True iff this file should be indexed.
+                      type: boolean
```
Comment (kislyuk) [inline dss-api.yml]: We may want to sniff/infer indexed status in the future (for example from content-type) but for now I'm fine either way.

File: tests/test_bundle.py [hunk 4]
```diff
@@ -30,6 +31,43 @@ def test_bundle_api(self):
...
+            ),
+        )
+        self.assertHeaders(
+            response[0],
+            {
+                'content-type': "application/json",
+            }
+        )
+        self.assertIn('version', response[2])
+
```
Comment (kislyuk) [inline tests/test_bundle.py]: Suggest adding some negative test cases when you have time.

Comment (ttung) [inline tests/test_bundle.py]: Will do.

**Comment by ttung** [comment]
Fixed the copy-pasta.

**Pull request merged**